File: evals/fixtures/express-ts/test_golden/src/models/store.ts
import { Guest, Property, Unit, Reservation } from "./types";

let idCounter = 0;

export function generateId(): string {
  idCounter++;
  return `id-${Date.now()}-${idCounter}`;
}

export const store = {
  guests: [] as Guest[],
  properties: [
    {
      id: "prop-1",
      name: "Sunset Rentals",
      address: "123 Sunset Blvd, Los Angeles, CA 90028",
    },
  ] as Property[],
  units: [
    { id: "unit-101", propertyId: "prop-1", name: "Unit 101" },
    { id: "unit-202", propertyId: "prop-1", name: "Unit 202" },
  ] as Unit[],
  reservations: [] as Reservation[],
};

export function findUnit(id: string): Unit | undefined {
  return store.units.find((u) => u.id === id);
}

export function findGuest(id: string): Guest | undefined {
  return store.guests.find((g) => g.id === id);
}

File: evals/fixtures/rails-rb/app/app/models/store.rb
module Store
  @id_counter = 0

  @guests = []

  @properties = [
    { id: "prop-1", name: "Sunset Rentals", address: "123 Sunset Blvd, Los Angeles, CA 90028" }
  ]

  @units = [
    { id: "unit-101", property_id: "prop-1", name: "Unit 101" },
    { id: "unit-202", property_id: "prop-1", name: "Unit 202" }
  ]

  @reservations = []

  class << self
    attr_accessor :guests, :properties, :units, :reservations

    def generate_id
      @id_counter += 1
      "id-#{Time.now.to_i}-#{@id_counter}"
    end

    def find_unit(id)
      @units.find { |u| u[:id] == id }
    end

    def find_guest(id)
      @guests.find { |g| g[:id] == id }
    end
  end
end
